pkgs/games/openra/default.nix

```diff
@@ -17,34 +18,82 @@ in stdenv.mkDerivation rec {
+    sha256 = "0dgaxy1my5r3sr3l3gw79v89dsc7179pasj2bibswlv03wsjgqbi";
+  };
+
+  sdl2_cs = fetchurl {
+    name = "SDL2-CS.dll";
+    url = "https://github.com/OpenRA/SDL2-CS/releases/download/20150709/SDL2-CS.dll";
+    sha256 = "0ms75w9w0x3dzpg5g1ym5nb1id7pmagbzqx0am7h8fq4m0cqddmc";
+  };
+
+  eluant = fetchurl {
+    name = "Eluant.dll";
+    url = "https://github.com/OpenRA/Eluant/releases/download/20140425/Eluant.dll";
+    sha256 = "1c20whz7dzfhg3szd62rvb79745x5iwrd5pp62j3bbj1q9wpddmb";
+  };
+
+  geolite = let filename = "GeoLite2-Country.mmdb.gz"; in
+    fetchurl {
+    name = filename;
+    url = "http://geolite.maxmind.com/download/geoip/database/${filename}";
+    sha256 = "0lr978pipk5q2z3x011ps4fx5nfc3hsal7jb77fc60aa6iscr05m";
   };
 
   dontStrip = true;
 
-  buildInputs = [ lua ];
+  buildInputs = with dotnetPackages;
+     [ NUnit NewtonsoftJson MonoNat FuzzyLogicLibrary SmartIrc4net SharpZipLib MaxMindGeoIP2 MaxMindDb SharpFont StyleCopMSBuild StyleCopPlusMSBuild RestSharp ]
+     ++ [ lua gnome3.zenity ];
   nativeBuildInputs = [ mono makeWrapper lua pkgconfig ];
 
   patchPhase = ''
     sed -i 's/^VERSION.*/VERSION = release-${version}/g' Makefile
-    substituteInPlace configure --replace /bin/bash "$shell" --replace /usr/local/lib "${lua}/lib"
+    substituteInPlace thirdparty/configure-native-deps.sh --replace "locations=\"" "locations=\"${lua}/lib "
+    substituteInPlace Makefile --replace "@./thirdparty/fetch-geoip-db.sh" ""
   '';
 
   preConfigure = ''
     makeFlags="prefix=$out"
     make version
   '';
 
+  preBuild = with dotnetPackages; ''
+    mkdir thirdparty/download/
+    cp ${MonoNat}/lib/dotnet/Mono.Nat/net40/Mono.Nat.dll thirdparty/download/
+    cp ${FuzzyLogicLibrary}/lib/dotnet/FuzzyLogicLibrary/Release/FuzzyLogicLibrary.dll thirdparty/download
+    cp ${SmartIrc4net}/lib/dotnet/SmartIrc4net/net40/SmarIrc4net* thirdparty/download/
+    cp ${SharpZipLib}/lib/dotnet/SharpZipLib/20/ICSharpCode.SharpZipLib.dll thirdparty/download/
+    cp ${MaxMindGeoIP2}/lib/dotnet/MaxMind.GeoIP2/net40/MaxMind.GeoIP2* thirdparty/download/
+    cp ${MaxMindDb}/lib/dotnet/MaxMind.Db/net40/MaxMind.Db.* thirdparty/download/
+    cp ${SharpFont}/lib/dotnet/SharpFont/net20/SharpFont.dll thirdparty/download/
+    cp ${SharpFont}/lib/dotnet/SharpFont/SharpFont.dll.config thirdparty/download/
+    cp ${StyleCopMSBuild}/lib/dotnet/StyleCop.MSBuild/StyleCop*.dll thirdparty/download/
+    cp ${StyleCopPlusMSBuild}/lib/dotnet/StyleCopPlus.MSBuild/StyleCopPlus.dll thirdparty/download
+    cp ${RestSharp}/lib/dotnet/RestSharp/net4-client/RestSharp.dll thirdparty/download
+    cp ${NUnit}/lib/dotnet/NUnit/nunit.framework.* thirdparty/download
+    cp ${NewtonsoftJson}/lib/dotnet/Newtonsoft.Json/Newtonsoft.Json.dll thirdparty/download
+    cp ${eluant} thirdparty/download/Eluant.dll
+    cp ${sdl2_cs} thirdparty/download/SDL2-CS.dll
+    cp ${geolite} "thirdparty/download/GeoLite2-Country.mmdb.gz"
+    make dependencies
```

Comment:
Done that for the dotnetPackages, the other ones need a rename, but I think this can be cleaned up a bit more. I need to read the nix docs again.